packages/db/tests/models/test-granule-model.js

```diff
@@ -402,6 +402,41 @@ test('GranulePgModel.upsert() will allow a running status to replace a non-runni
   t.is(record.status, 'running');
 });
 
+test('GranulePgModel.upsert() will allow a queued status to replace a non-queued status for different execution', async (t) => {
+  const {
+    knex,
+    executionPgModel,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    status: 'completed',
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  await upsertGranuleWithExecutionJoinRecord(knex, granule, executionCumulusId);
+
+  const [newExecutionCumulusId] = await executionPgModel.create(
+    t.context.knex,
+    fakeExecutionRecordFactory({ status: 'running' })
+  );
+
+  const updatedGranule = {
+    ...granule,
+    status: 'queued',
+  };
+
+  await granulePgModel.upsert(knex, updatedGranule, newExecutionCumulusId);
```

Comment:
Nit: can we use `upsertGranuleWithExecutionJoinRecord` for consistency?

```suggestion
  await upsertGranuleWithExecutionJoinRecord(knex, granule, newExecutionCumulusId);
```

Reply:
Since we're testing the `upsert` method here, shouldn't we call it directly?

Reply:
Yeah you're right. My bad.